pyirf/interpolation/base_interpolators.py

```diff
@@ -109,3 +117,10 @@ def __init__(self, grid_points, bin_edges, bin_contents):
         self.bin_edges = bin_edges
         self.bin_mids = bin_center(self.bin_edges)
         self.bin_contents = bin_contents
+
+    @classmethod
+    def __subclasshook__(cls, subclass):
```

Comment:
that's not really what I meant, although it works of course.

Did the documentation I link not work for some reason?

E.g. I expected this here after the class definition of the nearest neighbor classes:
```
DiscretePDFInterpolator.register(DiscretePDFNearestNeighborSearcher)
ParametrizedInterpolator.register(ParametrizedNearestNeighborSearcher)

```

Reply:
The link certainly did work, reading into this mechanic I merely thought that having this inside the class definition provides better overview. Is there an advantage in using `.register` over `__subclasshook__` or, directly asked, do you want me to change it to `.register`?

Reply:
Yes. If this lives inside the base class, users cannot do it for their implementation. The register approach allows anyone to add their own implementation. And then its about showing in our own code how we think it should be done.

Reply:
Another argument is that a base class shouldn't really have to know anything about its children. If we decide to remove or change `DiscretePDF...` we shouldn't need to modify the parent class.

Reply:
makes sense, changed